src/SharpCompress/IO/BufferedSubStream.cs

```diff
@@ -20,29 +20,41 @@ internal class BufferedSubStream(Stream stream, long origin, long bytesToRead)
 
     public override void Flush() { }
 
-    public override long Length => BytesLeftToRead;
+    public override long Length => BytesLeftToRead + _cacheLength - _cacheOffset;
```

Comment:
It seems weird that `Length` in a stream refers only to the remaining portion, but since that's the way it already worked I assume it's for a reason.

Reply:
maybe that's not the contract but I "meant" to do it 😬